[Refactor] Added HF model test

baberabb opened the pull request:
Have tested it locally ~~and will test it again here once the API key issue is resolved~~. Let me know if any of you have any feedback!

haileyschoelkopf commented:
Actually, it looks like the test is currently failing due to some roundoff. Maybe we could make loglikelihoods satisfy an allclose() rather than exact equality? That, or passing a set batch size to the LM() initialized here.

baberabb commented:
@haileyschoelkopf Sorry about that! Changed to allclose but not sure if atol of 0.01 is reasonable compared to those on my Mac? Also added a check for indices in case of multiple choice!

haileyschoelkopf commented:
That atol seems good for loglikelihood_rolling, but if a smaller one will still pass for loglikelihood, that might be ideal--if it still seems too flaky, then this is good for now.

baberabb commented:
Changed the tolerance to default for `loglikelihood` by getting the previous results from CI. Seems to be passing for now!

haileyschoelkopf commented:
Awesome! Feel free to merge whenever.

baberabb commented:
Good to merge!
